src/shared/dom.js

```diff
@@ -103,6 +103,7 @@ export function toNumber(value) {
 }
 
 export function timeRangesToArray(ranges) {
+	if (!ranges) return [];
```

Comment:
when could this be the case?

Reply:
Because the `src` attribute [may not be present](https://dev.w3.org/html5/spec-preview/media-elements.html#location-of-the-media-resource) and then all the ranges attributes return `undefined`.

Reply:
https://svelte.technology/repl?version=1.39.0&gist=66b93597de37579256dc333177c56e96

Reply:
In what browsers is that true? In Chrome:

<img width="297" alt="screen shot 2017-09-10 at 6 32 41 pm" src="https://user-images.githubusercontent.com/1162160/30253596-7882d0ea-9656-11e7-9c2b-e5e2c91802b6.png">

Fairly certain that's spec.

Reply:
I think what's happening here is we need to initialise the bindings to an empty array — adding the guard to this function probably won't fix it

Reply:
> we need to initialise the bindings to an empty array

Ah ok, I can do that.